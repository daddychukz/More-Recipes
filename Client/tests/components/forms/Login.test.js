import React from 'react';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ConnectedLogin, { PureLogin } from '../../../src/components/forms/LoginForm';

jest.mock('react-google-login');

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  auth: {
    isAuthenticated: false,
    user: {}
  },
  user: {}
};
const store = mockStore(initialState);

const props = {
  signIn: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  signUp: jest.fn(),
  resetPasswordRequest: sinon.spy(() => new Promise((cb) => {
    cb();
  }))
};

const mountedWrapper = mount(
  <BrowserRouter>
    <ConnectedLogin store={store} />
  </BrowserRouter>
);

const shallowWrapper = shallow(<PureLogin {...props} />);
const state = {
  Email: 'chuks@yahoo.com',
  Password: 'chuks'
};
const event = {
  preventDefault: jest.fn(),
  target: {
    Password: 'abcde',
    Email: 'chuks@yahoo.com',
  }
};
const googleResponse = {
  profileObj: {
    email: 'chuks@yahoo.com',
    name: 'chuks',
    googleId: 12345678,
    imageUrl: 'image.jpg'
  }
};

describe('Login', () => {
  it('renders Login component without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('LoginForm').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <PureLogin {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls sendResetLink event', () => {
    sinon.spy(shallowWrapper.instance(), 'sendResetLink');
    state.Email = 'abu@yahoo.com';
    shallowWrapper.setState(state);
    shallowWrapper.instance().sendResetLink(event);
    expect(shallowWrapper.instance().sendResetLink.calledOnce).toEqual(true);
  });

  it('calls GoogleSignup event', () => {
    sinon.spy(shallowWrapper.instance(), 'responseGoogle');
    shallowWrapper.instance().responseGoogle(googleResponse);
    expect(shallowWrapper.instance().responseGoogle.calledOnce).toEqual(true);
  });
});
