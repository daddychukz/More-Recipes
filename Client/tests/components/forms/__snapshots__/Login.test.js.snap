// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match component snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        name="Email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        name="Password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-info btn-block"
      type="submit"
      value="Login"
    />
    <br />
    <a
      data-target="#sendEmailModal"
      data-toggle="modal"
      href="#"
      onClick={[Function]}
    >
      <strong>
        forgot your password?
      </strong>
    </a>
    <hr
      className="bg-white"
    />
    or register with your Google account
    <ul
      className="list-inline text-center"
    >
      <li
        className="list-inline-item"
      >
        <a
          className="btn btn-lg"
          href="#"
          onClick={[Function]}
          title="Google"
        />
        Â  
      </li>
    </ul>
  </form>
  <div
    className="modal fade text-dark"
    id="sendEmailModal"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header bg-info"
        >
          <h5
            className="modal-title"
            id="contactModalTitle"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Forgot Password
          </h5>
        </div>
        <div
          className="modal-body"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="resetEmail"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <input
              className="btn btn-info btn-block"
              type="submit"
              value="Send Reset Link"
            />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
